intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -125,42 +120,53 @@ class MirrordConfigDropDown : ComboBoxAction() {
     // this function spawns a background task to query the index, not blocking the current thread
     // it maintains a single task to do so, using the `blockQueries` flag
     private fun queryIndexInSmartMode(project: Project, query: (Project) -> Unit) {
-        if (!blockQueries.getAndSet(true)) {
-            object : Task.Backgroundable(project, "mirrord") {
-                override fun run(indicator: ProgressIndicator) {
-                    val dumbService = DumbService.getInstance(project)
-                    // refer to `DumbService`, there is no "guarantee" still, but we stay in a loop
-                    // todo: should probably look into using the message bus to listen for indexing to finish
-                    while (true) {
-                        dumbService.waitForSmartMode()
-                        val success = ReadAction.compute<Boolean, RuntimeException> {
-                            if (project.isDisposed) {
-                                throw ProcessCanceledException()
-                            }
-                            if (dumbService.isDumb) {
-                                return@compute false
-                            }
-                            try {
-                                indicator.text = "mirrord: updating config files"
-                                query(project)
-                                blockQueries.set(false)
-                            } catch (e: IndexNotReadyException) {
-                                return@compute false
-                            }
-                            true
+        object : Task.Backgroundable(project, "mirrord") {
+            override fun run(indicator: ProgressIndicator) {
+                updateConfigs = false
```

Comment:
nitpick: I think it would be good to unset this flag as close as possible to the querying of the index, but up to you.